app/src/main/java/org/simple/clinic/textInputdatepicker/TextInputDatePickerEffectHandler.kt

```diff
@@ -18,6 +25,25 @@ class TextInputDatePickerEffectHandler @AssistedInject constructor(
         .subtypeEffectHandler<TextInputDatePickerEffect, TextInputDatePickerEvent>()
         .addAction(DismissSheet::class.java, { uiActions.dismissSheet() }, schedulersProvider.ui())
         .addAction(HideDateErrorMessage::class.java, { uiActions.hideErrorMessage() }, schedulersProvider.ui())
+        .addConsumer(ShowDateValidationError::class.java, { showDateValidationErrors(it.dateValidation) }, schedulersProvider.ui())
         .build()
   }
+
+  private fun showDateValidationErrors(dateValidation: Result) {
+    when (dateValidation) {
+      is InvalidPattern -> {
+        uiActions.showErrorMessage()
+        uiActions.showInvalidDateError()
```

Comment:
Yes we can but since these two have different actions, keeping them different for readability.

Reply:
In this scenario, it's fine because we won't be calling `showInvalidDateError` when the view is hidden right? So we can merge these 2 methods into one.

Reply:
Yes that is true but we'll be writing the same visibility property in all three functions if we use one method. Anyhow, I'll merge this into one method.